openg2p_registration/models/openg2p_registration.py

```diff
@@ -346,17 +369,83 @@ def create_beneficiary_from_registration(self):
             'context': context
         }
 
+    @api.onchange('stage_id')
+    def checking_duplicates(self):
+        if int(self.stage_id.id) == 2:
+            r = self.search_beneficiary()
+            if r == 200:
```

Comment:
200 doesn't gave the assurance that it'll be a duplicate. It is NOT a duplicate ONLY if the response array is empty which will return a 200 as well.